File: beecrowd/javascript/beginner/1036-bhaskaras-formula.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

const entrada = lines.shift();

const [A, B, C] = entrada.split(" ").map(coef => Number(coef));

const delta = B * B - 4 * A * C;

const divisor = 2 * A;

if (!divisor || delta < 0) {
  console.log("Impossivel calcular");
} else {
  const x1 = (-B + Math.sqrt(delta)) / divisor;
  const x2 = (-B - Math.sqrt(delta)) / divisor;
  console.log(`R1 = ${x1.toFixed(5)}`);
  console.log(`R2 = ${x2.toFixed(5)}`);
}

File: beecrowd/c++/beginner/1036-bhaskaras-formula.cpp
#include <iostream>
#include <iomanip>
#include <math.h>

using namespace std;

int main() {
  double a, b, c;

  cout << fixed << setprecision(5);

  cin >> a >> b >> c;

  double d = b * b - 4 * a * c;

  if (d < 0 || a == 0) {
    cout << "Impossivel calcular" << endl;
  } else {
    b = b * -1;
    d = sqrt(d);
    a *= 2;

    double x1 = b + d;
    x1 /= a;

    double x2 = b - d;
    x2 /= a;

    cout << "R1 = " << x1 << endl;
    cout << "R2 = " << x2 << endl;
  }

  return 0;
}
